src/script/components/RichTextEditor/plugins/EmojiPickerPlugin/EmojiPickerPlugin.tsx

```diff
@@ -238,10 +207,11 @@ export function EmojiPickerPlugin({openStateRef}: Props) {
   return (
     <TypeaheadMenuPlugin
       onQueryChange={setQueryString}
-      onSelectOption={onSelectOption}
+      onSelectOption={insertEmoji}
       triggerFn={checkForEmojiPickerMatch}
       options={options}
       menuRenderFn={menuRender}
+      onClose={() => (openStateRef.current = false)}
```

Comment:
This is the actual change. The rest is mostly cleaning up of unneeded operations